### ISSUE TITLE
Empty Errors

### ISSUE THREAD

**Issue by IBMRob** [packagesdev/packages#23]
We have migrated to the latest version of Packages and noticed that the build can often fail with no debug information making diagnosing the error really proplematic i.e.

```
[mouldy.:/myfolder/]/usr/local/bin/packagesbuild -v /build//temp/My_Enterprise.pkgproj
Building Project (19:04:39) at path:  /build//temp/My_Enterprise.pkgproj
------------------------------------------------------------------------------

Build Folder

==============================================================================
ERROR:

Description:


==============================================================================
Build Failed
```

**Comment by packagesdev** [comment]
A few questions:

1. Was the build working with the previous version you used?
2. Which previous version was it?
3. If you can launch a build from the application itself, are you getting more info about the error?

Based on the current step (Creating the Build Folder), it looks like the folder can not be created. But I don't see any error case not being reported with at least an explanation in the current code.

**Comment by IBMRob** [comment]
1. I suspect the failures are due to other changes on the configuration of the machine than the packages version. As we didn't have any failures (that I am aware of) we couldn't validate if this was or was not occurring with the old versions.

2. The previous version was 1.1.3

3. I have not managed to run the build as the same user via the UI as our automation does but if i run it manually using another ID it it does have a similar behaviour -  
   * If I just run a build it says the build failed but the drop down is empty with no reason.
   * If I "Build and debug" then the drop down is populated and it says permission denied (which is expected as my UI user doesn't have write permissions to the mount its trying to use)

**Comment by packagesdev** [comment]
is there a shared volume involved?

Between version 1.1.3 and version 1.2.2 I changed how the build process deals with the effective user id and group id. In version 1.1.3, if some kind of external/shared volumes was detected (e.g. an AppleShare volume), there would be some effective uid/gid switches to be able to read/write from/to the external/shared volumes.

In version 1.2.3 (the ones released this week), I addressed an issue where the supplemental groups were not set within the build process, therefore preventing for instance a user with admin privileges to create a package in a location with the following permissions root:admin 775.

So it might be worth checking whether the problem would not be solved with version 1.2.3.

**Comment by IBMRob** [comment]
Moving to 1.2.3 has changes it slightly. I am now getting an error code out:

```
[mouldy.:/myfolder/]/usr/local/bin/packagesbuild -v /build//temp/My_Enterprise.pkgproj
Building Project (20:02:39) at path:  /build//temp/My_Enterprise.pkgproj
------------------------------------------------------------------------------

Build Folder

==============================================================================
ERROR:

Description:

Error code (40)
==============================================================================
Build Failed
```

**Comment by packagesdev** [comment]
Error 40 occurs when the build folder already exists but is either actually a file or is not writable by the user who launched the build.

o A solution could probably be to remove the build folder manually. And try again.

o I will definitely add a more legible description of the error in the logs (app and command line).

**Comment by packagesdev** [comment]
Fixed in 1.2.4
